### PR TITLE
Fix to_int

### PR DESCRIPTION
**Pull request by faissaloo**

### CODE REVIEW

File: AK/String.cpp [hunk 1]
```diff
@@ -122,12 +122,30 @@ ByteBuffer String::to_byte_buffer() const
...
-// FIXME: Duh.
 int String::to_int(bool& ok) const
 {
-    unsigned value = to_uint(ok);
-    ASSERT(ok);
-    return (int)value;
+  bool negative = false;
+  int value = 0;
+  ssize_t i = 0;
+  if (characters()[0] == '-') {
```
Comment (awesomekling) [inline AK/String.cpp]: characters() may return null, so we must check that before using it like this.
The function could just start with
```
if (is_null()) {
    ok = false;
    return 0;
}
```

File: AK/String.cpp [hunk 2]
```diff
@@ -122,12 +122,30 @@ ByteBuffer String::to_byte_buffer() const
...
+      if (characters()[i] < '0' || characters()[i] > '9') {
+          ok = false;
+          return 0;
+      }
+      value = value * 10;
+      value += characters()[i] - '0';
+  }
+  ok = true;
+  
+  if (negative) {
```
Comment (awesomekling) [inline AK/String.cpp]: I would write this entire block as
```
return negative ? -value : value;
```

File: LibGUI/GVariant.h [hunk 3]
```diff
@@ -194,7 +194,7 @@ class GVariant {
         RawPoint as_point;
         RawSize as_size;
         RawRect as_rect;
-    } m_value; 
+    } m_value;
```
Comment (awesomekling) [inline LibGUI/GVariant.h]: This is an unrelated change and should not be in this commit.

Comment (faissaloo) [reply LibGUI/GVariant.h]: This was automatically done by my editor, I don't think it's worth excluding.

**Comment by faissaloo** [comment]
Should be good now

File: LibGUI/GVariant.h (hunk 3 above)
Comment (awesomekling) [reply LibGUI/GVariant.h]: Okay for this time, but in the future I'd prefer to not touch unrelated files since that makes the git history unnecessarily confusing :)

**Comment by awesomekling** [comment]
Awesome, thanks for fixing this! :+1:
